packages/viz/__tests__/styles/ColorCategoriesStyle.spec.ts

```diff
@@ -62,6 +67,54 @@ describe('ColorCategoriesStyle', () => {
         CartoStylingError
       );
     });
+
+    it('should fails with invalid top', () => {
+      const style = colorCategoriesStyle(FIELD_NAME, {
+        top: 0
```

Comment:
Using same reasoning as above, we should probably set the maximum top value allowed and be consistent on it. 

/style-helpers/color-categories-style.html doesn't fail with a high number of top, but it has only a low number of variations for the field being used. 

/examples/_old/styles/barcelona.html can be better for testing that limit. I made a quick test with top: 30 on it and it doesn't fail, but I think it does not properly assign colors, so IMO we should cap it too.

The qualitative palettes from cartocolors being used have a maximum length of 12 colors, so I guess that should be the limit too (valida range: 1 to 12)